go/test/endtoend/cluster/cluster_process.go

```diff
@@ -208,16 +214,26 @@ func (cluster *LocalProcessCluster) StartKeyspace(keyspace Keyspace, shardNames
 			log.Error(err.Error())
 			return
 		}
-
-		shards = append(shards, *shard)
+		keyspace.Shards = append(keyspace.Shards, *shard)
+	}
+	// if the keyspace is present then append the shard info
+	existingKeyspace := false
+	for idx, ks := range cluster.Keyspaces {
+		if ks.Name == keyspace.Name {
+			cluster.Keyspaces[idx].Shards = append(cluster.Keyspaces[idx].Shards, keyspace.Shards...)
+			existingKeyspace = true
+		}
+	}
+	if !existingKeyspace {
+		cluster.Keyspaces = append(cluster.Keyspaces, keyspace)
 	}
-	keyspace.Shards = shards
-	cluster.Keyspaces = append(cluster.Keyspaces, keyspace)
 
 	// Apply Schema SQL
-	if err = cluster.VtctlclientProcess.ApplySchema(keyspace.Name, keyspace.SchemaSQL); err != nil {
-		log.Error(err.Error())
-		return
+	if keyspace.SchemaSQL != "" {
+		if err = cluster.VtctlclientProcess.ApplySchema(keyspace.Name, keyspace.SchemaSQL); err != nil {
+			log.Error(err.Error())
```

Comment:
Why doesn't the test fail on this error?

Reply:
This is just in case, the applied sql schema is not correct.

Reply:
This is just in case the sql schema is not correct. In the test, I invoked this separately when I pass incorrect sql schema. But at this line it always gets correct sql schema.

Reply:
I still think the framework should fail if ApplySchema fails. That will be useful when writing tests, otherwise it will be harder to diagnose.

Reply:
Yes, you are right. In case of error, it returns `err` and wherever `StartKeyspace` is called, if that gets error it return with `exitcode` as 1.